### ISSUE TITLE
ObjC cannot see methods made with @implementation

### ISSUE THREAD

**Issue by bnovc** [frida/frida#37]
Methods that are added to a class with @implementation in ObjC cannot be seen or attached to by Frida.

e.g.
import frida                                                                                                                                                                                                                                                          
import sys                                                                                                                                                                                                                                                            
                                                                                                                                                                                                                                                                      
session = frida.attach(455)        // some process with a menu bar icon                                                                                                                                                                                                                                   
                                                                                                                                                                                                                                                                                                                                                                                                                                                               
script = session.create_script("""                                                                                                                                                                                                                                    
var NSSound = ObjC.use('NSStatusItem'); /* Mac */                                                                                                                                                                                                                     
var oldImpl = NSSound.drawRect.implementation;                                                                                                                                                                                                                        
NSSound.drawRect.implementation = ObjC.implement(NSSound.drawRect, function (handle, selector) {                                                                                                                                                                      
  console.log('drawrect');                                                                                                                                                                                                                                            
  return oldImpl(handle, selector);                                                                                                                                                                                                                                   
});                                                                                                                                                                                                                                                                   
""");                                                                                                                                                                                                                                                                 
                                                                                                                                                                                                                                                                      
def on_message(message, data):                                                                                                                                                                                                                                        
  print "{}, {}".format(message, data)                                                                                                                                                                                                                                
                                                                                                                                                                                                                                                                      
script.on('message', on_message)                                                                                                                                                                                                                                      
script.load()                                                                                                                                                                                                                                                         
sys.stdin.read()

**Comment by oleavr** [comment]
Hmm, perhaps you meant `drawRect_` for `drawRect:`? Frida 4.0.0 has just been released with a fully revamped ObjC integration, so it would be great if you could give it a spin.

The API is now:
```JavaScript
var NSStatusItem = ObjC.classes.NSStatusItem;
var method = NSStatusItem["- drawRect:"];
method.implementation = ObjC.implement(method, function (handle, selector) {
});
```

More examples [here](https://github.com/frida/frida-gum/blob/cf6584724c684c9366bd8051d2148022b5606773/tests/gumjs/script-mac.m).

Cheers!

**Comment by oleavr** [comment]
Considering this one fixed.

**Comment by bnovc** [comment]
You can see them now but you still can't change their implementation.

**Comment by oleavr** [comment]
Fixed :) (4.1.9 is out with the fix)
